src/python/dxpy/cli/dataset_utilities.py

```diff
@@ -1047,6 +1058,199 @@ def print_fields(fields):
+            "Original Error message:\n{}"
+        ).format(str(exc))
+    # Order of samples doesn't matter so using set here
+    discovered_ids = set()
+    # Parse the results objects for the cohort ids
+    for result in resp_raw["results"]:
+        discovered_ids.add(result[field_name])
+
+    # Compare the discovered cohort ids to the user-provided cohort ids
+    if discovered_ids != set(ids):
+        # Find which given samples are not present in the dataset
+        missing_ids = set(ids).difference(discovered_ids)
+        err_msg = "The following supplied IDs do not match IDs in the main entity of dataset, {dataset_name}: {ids}".format(dataset_name = resp["dataset"],ids = missing_ids)
+        raise ValueError(err_msg)
+
+        
+def has_access_level(project, access_level):
+    """
+    Validates that issuing user has required access level.
+    Args:
+        project: str: tasked project_id
+        access_level: str: minimum requested level
+    Retuns: boolean
+    """
+    level_rank = ["VIEW", "UPLOAD", "CONTRIBUTE", "ADMINISTER"]
+    access_level_idx = level_rank.index(access_level)
+    try:
+        project_describe = describe(project)
+    except PermissionDenied:
+        return False
+    if level_rank.index(project_describe["level"]) < access_level_idx:
+        return False
+    return True
+
+
+def validate_project_access(project, access_level="UPLOAD"):
+    """
+    Validates that project has requested access.
+    Args:
+        project: str: tasked project_id
+        access_level: str: minimum requested level Default at least UPLOAD
+    Returns:
+        Error message
+
+    """
+    if not has_access_level(project, access_level):
+        return "At least {} permission is required to create a record in a project".format(
+            access_level
+        )
+
+
+def create_cohort(args):
+    """
+    Create a cohort from dataset/cohort and specified list of samples.
+    """
+    #### Validation ####
+    # Validate and resolve 'PATH' input
+    
+    # default path values
+    path_project = dxpy.WORKSPACE_ID
+    path_folder = dxpy.config.get('DX_CLI_WD', '/')
+    path_name = None
+    if args.PATH:
+        path_project, path_folder, path_name, err_msg = resolve_validate_dx_path(
+            args.PATH
+        )
+        if err_msg:
+            err_exit(err_msg)
+        err_msg = validate_project_access(path_project)
+        if err_msg:
+            err_exit(err_msg)
+    # validate and resolve 'from' input
+    FROM = args.__dict__.get("from")
+    from_project, entity_result, resp, dataset_project = resolve_validate_record_path(FROM)
+
+    #### Reading input sample ids from file or command line ####
+    samples=[]
+    # from file
+    if args.cohort_ids_file:
+        with open(args.cohort_ids_file,"r") as infile:
+            for line in infile:
+                samples.append(line.strip("\n"))
+    # from string
+    if args.cohort_ids:
+        samples = args.cohort_ids.split(",")
```

Comment:
Will it be documented that there cannot be any spaces between sample ids?

Reply:
Good point. I think we should add support for spaces between sample IDs.